### PR TITLE
Barsign UI/UX Improvements 

### PR DESCRIPTION
**Pull request by Chirynnn**
## About The Pull Request
This ports https://github.com/tgstation/tgstation/pull/73106, but no emissives. They do not work. This is part of the Things to Port list.

## Why It's Good For The Game
Barsigns not being a machine makes no sense.

## Testing Photographs and Procedure

https://user-images.githubusercontent.com/77589317/226995958-78076374-ee7f-433e-a725-ebde530f9347.mp4

<details>
<summary>Screenshots&Videos</summary>

</details>

## Changelog
:cl: timothymtorres, Chirynnn
tweak: Add balloon alerts to barsigns. Barsigns are now considered machinery and will be affected by power.
refactor: Refactored barsign code
/:cl:

### CODE REVIEW

**Comment by itsmeow** [comment]
emissives do work we just have a different implementation

**Comment by Chirynnn** [comment]
RKz couldn't do it either, so I don't know. So is this not getting merged without them?

File: code/game/machinery/barsigns.dm [hunk 1]
```diff
@@ -1,133 +1,178 @@
...
-		if(initial(D.name) == sign_name)
-			var/new_sign = new D
-			return set_sign(new_sign)
+/obj/machinery/barsign/update_appearance(updates=ALL)
+	. = ..()
+	if(!(machine_stat & (NOPOWER|BROKEN)))
+		set_light(0)
+		return
+	if(chosen_sign && chosen_sign.neon_color)
+		set_light(MINIMUM_USEFUL_LIGHT_RANGE, 0.7, chosen_sign.neon_color)
```
Comment (Crossedfall) [inline code/game/machinery/barsigns.dm:72]: This is always going to return with `set_light(0)` as `machine_stat` is going to be 0.

File: code/game/machinery/barsigns.dm [hunk 2]
```diff
@@ -1,133 +1,178 @@
...
+	. = ..()
 
-	if(sign.rename_area && sign.name)
-		rename_area(src, sign.name)
+	if(!(machine_stat & (NOPOWER|BROKEN)))
+		return
 
-	return sign
+	if(chosen_sign && chosen_sign.light_mask)
+		. += emissive_appearance(icon, "[chosen_sign.icon]-light-mask", src)
```
Comment (Crossedfall) [inline code/game/machinery/barsigns.dm:64]: This wont trigger because it'll return with the `machine_stat` check.

**Review by Crossedfall — COMMENTED** [review]
`machine_stat` is going to be 0 pretty much all the time (edit: with the current changes*) so a lot of the added stuff isn't going to ever fire. That includes the emissive lights. If you comment out the two `machine_stat` checks in `update_overlays` and `update_appearance` you should see the emissive effect. 

As Kat was pointing out, our current implementation is different to TG's, so, you're not going to get the same striking appearance. It's going to be similar to vending machines where there's a soft glow coming from behind the object.

**Comment by ike709** [comment]
Stale PR is stale.

**Pull request closed without merging**